activestorage/lib/active_storage/service/disk_service.rb

```diff
@@ -72,6 +72,10 @@ def exist?(key)
       end
     end
 
+    def move(source_key, target_key)
+      FileUtils.mv(path_for(source_key), path_for(target_key), force: true)
```

Comment:
Playing around with this I discovered that this will fail if the destination directory does not exist. My solution:
```suggestion
      FileUtils.mkdir_p(File.dirname(path_for(target_key)))
      FileUtils.mv(path_for(source_key), path_for(target_key))
```